### PR TITLE
ENH: py2 compat wrappers

### PR DESCRIPTION
**Pull request by llllllllll**

### CODE REVIEW

File: interface/compat.py [hunk 1]
```diff
@@ -6,17 +7,42 @@
 PY2 = version_info.major == 2
 PY3 = version_info.major == 3
 
-if PY2:  # pragma: nocover
+if PY2:  # pragma: nocover-py3
```
Comment (ssanderson) [inline interface/compat.py]: :+1:

File: interface/compat.py [hunk 2]
```diff
@@ -6,17 +7,42 @@
...
     from funcsigs import signature, Parameter
 
+    @functools.wraps(functools.wraps)
```
Comment (ssanderson) [inline interface/compat.py]: :thinking:

File: interface/tests/test_interface.py [hunk 3]
```diff
@@ -656,3 +656,21 @@ def default_classmethod(cls, x):
...
+
+    def wrapping_decorator(f):
+        @wraps(f)
+        def inner(*args, **kwargs):  # pragma: nocover
+            pass
+
+        return inner
+
+    class C(implements(I)):  # pragma: nocover
+        @wrapping_decorator
```
Comment (ssanderson) [inline interface/tests/test_interface.py]: Can we also test that we fail correctly here if the impl signature is incorrect?

Comment (llllllllll) [reply interface/tests/test_interface.py]: sure

**Comment by ssanderson** [comment]
LGTM

**Pull request merged**